pkg/pagerduty/pagerduty.go

```diff
@@ -62,7 +62,38 @@ func (p PagerDuty) MoveToEscalationPolicy(incident Incident, escalationPolicy Es
 	}
 	err := p.manageIncident(o)
 	if errors.Is(err, FailedToUpdateIncidentError{}) {
-		return pkgerrors.Wrap(err, "could not update the escalation policy")
+		return fmt.Errorf("could not update the escalation policy: %w", err)
+	}
+	return err
+}
+
+// AssignToCurrentUser will assign the incident to the current logged in user
+func (p PagerDuty) AssignToCurrentUser(incident Incident) error {
+	err := p.AssignToUser(incident, p.currentUser)
+	if err != nil {
+		return fmt.Errorf("could not assign to the current user: %w", err)
+	}
+	return nil
+}
+
+// AssignToUser will assign the incident to the current user
+func (p PagerDuty) AssignToUser(incident Incident, user User) error {
+	o := []sdk.ManageIncidentsOptions{
+		{
+			ID: incident.ID,
+			Assignments: []sdk.Assignee{
+				{
+					Assignee: sdk.APIObject{
+						Type: "user_reference",
+						ID:   user.ID,
+					},
+				},
+			},
+		},
+	}
+	err := p.manageIncident(o)
+	if errors.Is(err, FailedToUpdateIncidentError{}) {
```

Comment:
again like above, I would probably keep it generic with:


```suggestion
	if err != nil {
```
As InvalidTokenError and InvalidInputParameterError have the same effect, meaning `"could not assign to user"` And you should also state that in their error messages.